### PR TITLE
fixed smart indentation\formatting in template literals

### LINKED ISSUE #1751: Indentation is added when Enter is pressed inside template literal
```typescript
var x = `asd$asd`
```
press Enter at $ - rest of template literal will be moved to the next line (OK) and indented (WRONG)

### PR DESCRIPTION
**Pull request by vladima**
Fixes #1751

### CODE REVIEW

File: src/services/formatting/smartIndenter.ts [hunk 1]
```diff
@@ -12,10 +12,15 @@ module ts.formatting {
...
-                precedingToken.getStart(sourceFile) <= position &&
-                precedingToken.end > position) {
+            // no indentation in string \regex\template literals
+            var precedingTokenIsLiteral =
+                precedingToken.kind === SyntaxKind.StringLiteral ||
+                precedingToken.kind === SyntaxKind.RegularExpressionLiteral ||
+                precedingToken.kind === SyntaxKind.NoSubstitutionTemplateLiteral ||
+                precedingToken.kind === SyntaxKind.TemplateHead ||
+                precedingToken.kind === SyntaxKind.TemplateMiddle ||
+                precedingToken.kind === SyntaxKind.TemplateTail;
```
Comment (DanielRosenwasser) [inline src/services/formatting/smartIndenter.ts]: Use `isTemplateKind`

File: tests/cases/fourslash/formatTemplateLiteral.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/// <reference path="fourslash.ts"/>
+////var x = `sadasdasdasdasfegsfd
+/////*1*/rasdesgeryt35t35y35 e4 ergt er 35t 3535 `;
+////var y = `1${2}/*2*/3`;
```
Comment (mhegazy) [inline tests/cases/fourslash/formatTemplateLiteral.ts]: do we have a test for indentation inside the expression part of the template?

**Comment by mhegazy** [comment]
:+1:

**Comment by mhegazy** [comment]
also can we log an issue to track the change.

**Pull request merged**